p2p/p2p.go

```diff
@@ -264,24 +280,51 @@ func generateKey(seed int64) (crypto.PrivKey, error) {
 	return priv, nil
 }
 
-// TODO: message handling
+// handles stream; reads message length, message type, and decodes message based on type
+// TODO: implement all message types; send message back to peer when we get a message; gossip for certain message types
 func handleStream(stream net.Stream) {
 	defer func() {
 		if err := stream.Close(); err != nil {
 			log.Error("error closing stream", "err", err)
 		}
 	}()
-	// Create a buffer stream for non blocking read and write.
+
 	rw := bufio.NewReadWriter(bufio.NewReader(stream), bufio.NewWriter(stream))
-	str, err := rw.ReadString('\n')
+	lengthByte, err := rw.Reader.ReadByte()
```

Comment:
The logic from here till line 306 shouldn't be part of the stream handler, whatever is responsible for decoding the message should handle this

Reply:
okay I've gotten rid of this added a DecodeMessage func which decodes it into the right message type